lib/galaxy/webapps/galaxy/services/workflows.py

```diff
@@ -124,6 +124,8 @@ def invoke_workflow(
         workflow_id,
         payload: InvokeWorkflowPayload,
     ) -> Union[WorkflowInvocationResponse, List[WorkflowInvocationResponse]]:
+        if trans.anonymous:
+            raise exceptions.AuthenticationRequired("You need to be logged in to run workflows.")
```

Comment:
```suggestion
            raise exceptions.AuthenticationRequired("You need to be logged in to run workflows.")
        trans.check_user_activation()
```

Reply:
Ah nice, this way we check inactive users AND anonymous users. We can just do this then, or do you think doing what I recommended in my reply to Marius above is better?

Reply:
This is better, thanks and sorry for the confusion!

Reply:
> Ah nice, this way we check inactive users AND anonymous users. We can just do this then, or do you think doing what I recommended in my reply to Marius above is better?

I suggested to keep them separate to have different error messages.

Reply:
Yes very good point!